lib/restforce/concerns/composite_api.rb

```diff
@@ -23,22 +23,21 @@ def composite(all_or_none: false, collate_subrequests: false)
           collateSubrequests: collate_subrequests
         }
         response = api_post('composite', properties.to_json)
+        response.body['compositeResponse']
```

Comment:
Isn't the purpose of this PR to return the entire response body when raising an error? This change doesn't seem to solve that issue as line 36 hasn't changed.

Reply:
there were 3 ways to go about this:
1. Return `CompositeAPIError` with entire error message as a string. The error message is an Array of Hash here so it needs to be serialized into string before forming `CompositeAPIError`
   - **Tradeoff**:  Error message needs to be serialized  by SDK & de-serialized by Client.
2. Extend `CompositeAPIError` to intake extra field to store error message.
   - **Tradeoff** : `CompositeAPIError` is now different error from rest of the other error object which is inconsistent. 
3. Just return entire `compositeResponse` in `complaint` & throw `CompositeAPIError` in `complaint!` if `allOrNone` is true & `compositeResponse` has error.
   -  **Tradeoff**:  Client needs to process error on its own when using `complaint` or rely on `complaint!` to throw error. 
 
With all the approach listed above, I personally feel approach 3 is cleaner. Every use case will be different and having the option to return a raw `compositeResponse` opens up client to their interpretation of when to raise the error.

Reply:
sorry I just realized a major flaw with approach 3 - it is a major breaking change to existing SDK users. I think approach 2 is the best way forward without breaking the existing functionality. I will update this PR.